server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -1345,6 +1347,25 @@ func (s *BuildBuddyServer) InvalidateSnapshot(ctx context.Context, request *wfpb
 	return nil, status.UnimplementedError("Not implemented")
 }
 
+func (s *BuildBuddyServer) InvalidateAllSnapshotsForRepo(ctx context.Context, req *wfpb.InvalidateAllSnapshotsForRepoRequest) (*wfpb.InvalidateAllSnapshotsForRepoResponse, error) {
+	if wfs := s.env.GetWorkflowService(); wfs != nil {
+		if err := s.checkInvalidateSnapshotPerms(ctx); err != nil {
+			return nil, status.UnauthenticatedError(err.Error())
+		}
+
+		groupID := req.GetRequestContext().GetGroupId()
```

Comment:
nit: the preferred way to get the group ID is `u, err := env.GetAuthenticator().GetAuthenticatedUser()` then call `u.GetGroupID()`. The auth layer takes care of determining the group ID based on request context + JWT for web requests, or API key owner for API requests.

(also missed this in the RPC added above).

It'd also be good to avoid passing the raw `groupID` to `InvalidateAllSnapshotForRepo` and get the group ID inside that func instead - generally we want to avoid passing around raw group IDs as much as possible.

Reply:
Done